### PR TITLE
Leafref addtions to vrf and peer_group attributes in Sonic BGP Yangs.

### PR DESCRIPTION
**Pull request by prvattem**

### CODE REVIEW

**Comment by prvattem** [comment]
Yangs to review:
models/yang/sonic/sonic-bgp-common.yang
models/yang/sonic/sonic-bgp-global.yang 
 models/yang/sonic/sonic-bgp-neighbor.yang
models/yang/sonic/sonic-bgp-peergroup.yang
models/yang/sonic/sonic-route-common.yang

**Review by dutta-partha — APPROVED** [review]
Looks good.

File: models/yang/sonic/sonic-bgp-neighbor.yang [hunk 1]
```diff
@@ -48,8 +59,21 @@ module sonic-bgp-neighbor {
...
+                }
+
                 leaf neighbor_ip {
-                    type inet:ip-address;
+                    type leafref {
+                        path "../../../BGP_NEIGHBOR/BGP_NEIGHBOR_LIST/neighbor_ip";
+                    }
+
+                    must "/sbnbr:sonic-bgp-neighbor/sbnbr:BGP_NEIGHBOR/sbnbr:BGP_NEIGHBOR_LIST[vrf_name=current()/../vrf_name][neighbor_ip=current()]" {
+                            error-message "BGP Neighbor Table is not configured with vrf_name and neighbor_ip";
```
Comment (mhampasagar) [inline models/yang/sonic/sonic-bgp-neighbor.yang:84]: Other sonic YANGS has error like below, may be we can follow the same
error-message "Invalid Instance";
error-app-tag instance-number-invalid;

File: models/yang/sonic/sonic-bgp-common.yang [hunk 2]
```diff
@@ -68,16 +68,6 @@ module sonic-bgp-common {
 
   grouping sonic-bgp-cmn {
 
-    leaf vrf_name {
-      type string;
-        description "Network-instance/VRF name";
-    }
-
-    leaf peer_group_name {
```
Comment (mhampasagar) [inline models/yang/sonic/sonic-bgp-common.yang:76]: As peer_group_name is common to both NBR and peer group, looks like its missed to add it in nbr table

Comment (prvattem) [reply models/yang/sonic/sonic-bgp-common.yang:76]: Added this attribute to SONIC_BGP_NEIGHBOR table

**Review by joyas-joseph — APPROVED** [review]

**Pull request merged**